### PR TITLE
feat(ast) Remove dependency on legacy query from splitter

### PR DESCRIPTION
**Pull request by fpacifici**
Three of the four metrics we collected around discrepancies in managing the legacy query on query splitter turned out to be expected. So removing that log and the related dependency.

The fourth seems fine (actually it was a bug in the legacy system) but I need to do one last verification, thus logging the stack trace.

### CODE REVIEW

File: snuba/web/split.py [hunk 1]
```diff
@@ -159,14 +159,9 @@ def execute(
             logger.warning(
                 "Mismatch in start date on time splitter.",
                 extra={"ast": str(from_date_ast), "legacy": str(from_date)},
+                exc_info=True,
             )
             metrics.increment("mismatch.ast_from_date")
-        if to_date != to_date_ast:
-            logger.warning(
```
Comment (fpacifici) [inline snuba/web/split.py:165]: Never saw this message being logged. So there is no discrepancy here.

File: snuba/web/split.py [hunk 2]
```diff
@@ -290,22 +285,13 @@ def execute(
...
         # not the number of distinct Column objects in the query
         # so to avoid counting aliased columns multiple times.
-        total_ast_cols = {
+        total_columns = {
             (col.table_name, col.column_name)
             for col in query.get_all_ast_referenced_columns()
         }
-        total_ast_count = len(total_ast_cols)
-        if total_col_count != total_ast_count:
-            logger.warning(
```
Comment (fpacifici) [inline snuba/web/split.py:304]: This was a limitation of the legacy query. It was counting literals as columns. So the right result is the one from the AST

File: snuba/web/split.py [hunk 3]
```diff
@@ -340,21 +326,11 @@ def execute(
...
-        minimal_ast_cols = {
+        minimal_columns = {
             (col.table_name, col.column_name)
             for col in minimal_query.get_all_ast_referenced_columns()
         }
-        minimal_ast_count = len(minimal_ast_cols)
-        minimal_legacy_cols = minimal_query.get_all_referenced_columns()
-        minimal_count = len(minimal_legacy_cols)
-        if minimal_count != minimal_ast_count:
-            logger.warning(
```
Comment (fpacifici) [inline snuba/web/split.py:351]: This was a limitation of the legacy query. It was counting literals as columns. So the right result is the one from the AST

File: snuba/web/split.py [hunk 4]
```diff
@@ -159,14 +159,9 @@ def execute(
             logger.warning(
                 "Mismatch in start date on time splitter.",
                 extra={"ast": str(from_date_ast), "legacy": str(from_date)},
+                exc_info=True,
```
Comment (fpacifici) [inline snuba/web/split.py:162]: This seems due to the fact that the legacy query parsing just looks for the first occurrence of a date condition in the query, while the ast (correctly) looks for the most recent datetime. So the ast is providing the right result.

Logging the stack trace before changing the behavior just to verify this on more than a few queries.

**Review by tkaemming — APPROVED** [review]

**Pull request merged**